lib/librato-services/output.rb

```diff
@@ -143,6 +143,28 @@ def calculate_duration(measurement)
         end
       end
 
+      def sms_message
+        if valid_sms?
+          violations_message
+        else
+          truncated_violations_message
+        end
+      end
+
+      def valid_sms?
+        violations_message.length <= 140
+      end
+
+      def violations_message
+        violations.flat_map do |source, measurements|
+          measurements.map { |measurement| format_measurement(measurement, source) }
+        end.join('. ')
+      end
+
+      def truncated_violations_message
+        "#{violations_message[0..136]}..."
```

Comment:
This could be a general-purpose truncation method if the allowed length was a parameter (which can then be passed in from an SMS-specific method). 

Also, you can fit two additional characters if you use … (the ellipsis character).